File: app/src/main/java/com/example/rhythmgame2024/GameLoop.java
package com.example.rhythmgame2024;

import android.util.Log;

public class GameLoop  implements Runnable{

    private Thread gameThread;
    private GamePanel gamePanel;

    public boolean isRunning = true;

    public GameLoop(GamePanel gamePanel){
        this.gamePanel = gamePanel;
        gameThread = new Thread(this);
    }

    @Override
    public void run(){

        while(isRunning){
            gamePanel.update(1F);
            gamePanel.render(0F);
            //gamePanel.beatListUpdate();
        }
    }

    public void startGameLoop(){
        if(!gameThread.isAlive()) {
            gameThread.start();
        }
    }

    public void endGameLoop(){
        isRunning = false;
        System.out.println("game ends");
    }

}

File: app/src/main/java/com/example/rhythmgame2024/GameLoop.kt
package com.example.rhythmgame2024

import android.util.Log
import kotlin.math.log

class GameLoop(gamePanel: GamePanel) : Runnable{
    private lateinit var gamePanel : GamePanel
    private val gameThread: Thread = Thread(this)


    override fun run() {
        while (true){
            Log.d("GameLoop", "run: gameLoop rendering")
            gamePanel.render()
        }
    }

    fun startGameLoop(){
        gameThread!!.start()
        Log.d("GameLoop", "startGameLoop: true")
    }
}
